bioptim/optimization/receding_horizon_optimization.py

```diff
@@ -129,14 +138,14 @@ def solve(
             skip_continuity=True,
         )
 
-        states = InitialGuess(np.concatenate(states, axis=1), interpolation=InterpolationType.EACH_FRAME)
-        controls = InitialGuess(np.concatenate(controls, axis=1), interpolation=InterpolationType.EACH_FRAME)
-        sol = Solution(solution_ocp, [states, controls])
+        self.states = InitialGuess(np.concatenate(self.states, axis=1), interpolation=InterpolationType.EACH_FRAME)
+        self.controls = InitialGuess(np.concatenate(self.controls, axis=1), interpolation=InterpolationType.EACH_FRAME)
```

Comment:
Why is that needed to put these in self? That does not feel right to save data in the MHE class. Why don't you just use `sol.states["all"]` if you need the state values?

Reply:
This has been removed